### ISSUE TITLE
Trigger integration "SalesForce" thorugh the CP

### ISSUE THREAD

**Issue by Romanavr** [verbb/formie#1243]
### Describe the bug

We have integration with "SalesForce" through your plugin.
On the front end all forms work as expected & leads are successfully delivered.
However, for some submissions we need to push leads through admin panel using button in the screenshot.
https://i.imgur.com/TAaFoM1.png

After pressing the button, the flash notification says that everything worked successfully, but in fact it looks like it didn't.
First of all, I don't see the lead in the salesforce.
Also every time a submission fired on the frontend - regardless of the success I can see it in the log file (formie-XXX.log), but not in this case.
Moreover, I did some research with debugging, and from my point of view, it looks like the payload for the API is wrong?
Even if I click this button in the submissions with an existing lead, it says everything is fine, but it shouldn't be because the plugin should throw an API error, which in this case is a duplicate lead.

Maybe I'm just missing something.

### Steps to reproduce

1. Go to the admin panel(CP),
2. Select any submission
3. Try to trigger integration with the salesforce thorugh button "Intergrations"
4. Even if the lead is duplicated, the plugin will say that everything is in order.

### Form settings

- Multi-page form: Yes
- Submission Method: Ajax
- Client-side Validation: No
- Custom Form Templates: No

### Craft CMS version

4.3.6

### Plugin version

2.0.20

### Multi-site?

Yes

**Comment by engram-design** [comment]
It's true that the message would always show as successful, but it should still log any errors to the log file, if one occurs.

I'm not sure if you have the **Create Task for Duplicates** setting enabled? That largely depends on your Salesforce duplicate settings, which in some cases can allow duplicates. That may determine whether this is technically an error or not.

However, it looks like we're not properly reporting the error state of Salesforce leads, so that may likely be the issue. Fixed for the next release. To get this early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-4 as 2.0.20",
  "...": "..."
}
```

Then run `composer update`.

**Comment by engram-design** [comment]
Fixed in [2.0.21](https://github.com/verbb/formie/releases/tag/2.0.21)
